src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -507,46 +520,66 @@ private static bool TryParseValue(ref Utf8JsonReader reader, [NotNullWhen(true)]
             }
 
             int length = valueSpan.IsEmpty ? checked((int)valueSequence.Length) : valueSpan.Length;
-            byte[] rented = ArrayPool<byte>.Shared.Rent(length);
-            Span<byte> rentedSpan = rented.AsSpan(0, length);
+            if (useArrayPools)
+            {
+                byte[] rented = ArrayPool<byte>.Shared.Rent(length);
+                Span<byte> rentedSpan = rented.AsSpan(0, length);
 
-            try
+                try
+                {
+                    if (valueSpan.IsEmpty)
+                    {
+                        valueSequence.CopyTo(rentedSpan);
+                    }
+                    else
+                    {
+                        valueSpan.CopyTo(rentedSpan);
+                    }
+
+                    document = Parse(rented.AsMemory(0, length), state.Options, rented, useArrayPools);
+                }
+                catch
+                {
+                    // This really shouldn't happen since the document was already checked
+                    // for consistency by Skip.  But if data mutations happened just after
```

Comment:
```suggestion
                    // for consistency by Skip. But if data mutations happened just after
```

Reply:
The double space after a period appears to be a pattern in this code base -- @bartonjs?

Reply:
I'm old, so I initially learned to type on a typewriter, and double-space-after-period is built into my hands. (And, personally, I think generally looks better in fixed with fonts, which is why the concept was introduced in the first place).

I'd leave it as is, since it's just an indentation/move that has no other formatting change to it (whether it matched or didn't match the style).  If you want my style opinion, clearly it's double-space-after-period.  But if you're merely asking if it'll make me annoyed if it gets changed: nah :smile:.

Reply:
I don't plan on changing. Thanks. If someone wants to make the spacing consistent (and any other things) then yes that's fine but should be in a separate PR.